tf2onnx/optimizer/back_to_back_optimizer.py

```diff
@@ -83,7 +83,7 @@ def _optimize_cast(g, node, consumer_nodes):
             type2 = pnode.get_attr('to').i
             if type1 == type2:
                 for node2 in consumer_nodes:
-                    node2.input[0] = node.input[0]
```

Comment:
If we want to keep the old syntax, you could just let input be a subclass of list that tracks when its elements are changed and updates the cache.  But this works too.

Reply:
We could do that. I chose to replace this syntax by a method because users usually do not expect this syntax to do anything complex such as calling a method of the graph owning this node and updating some indices. And the documentation of the method is easier to find with an explicit name.

Reply:
Ah, makes sense.